packages/core/src/modules/proofs/services/ProofService.ts

```diff
@@ -369,8 +373,10 @@ export class ProofService {
     }
     await validateOrReject(proofRequest)
 
-    // Assert attribute and predicate group names do not match
+    // Assert attribute and predicate (group) names do not match
     this.assertAttributePredicateGroupNamesDoNotMatch(proofRequest)
+    this.assertAttributeNamesDoNotMatch(proofRequest)
+    this.assertAttributePredicateNamesDoNotMatch(proofRequest)
```

Comment:
I think one method would suffice for all three checks (although it might feel bit hacky)

```ts
const assertDuplicatesInArray(arr: string[]): void {
  const arrayLength = arr.length
  const uniqueArrayLength = new Set(arr).size

  // Minor optimalization
  if(arrayLength === uniquearrayLength) return;

  const duplicates = arr.filter((item, index) => arr.indexOf(item) != index)
  throw new AriesFrameworkError(`The proof request contains a duplicate items: ${duplicates.toString()}`)
}

// And call like so

// replicate assertAttributeNamesDoNotMatch()
 const attributes = Array.from(proofRequest.requestedAttributes.values()).map((a) => a.name)
assertDuplicatesInArray(attributes)

// replicate assertPredicateNamesDoNotMatch()
const predicates = Array.from(proofRequest.requestedPredicates.values()).map((a) => a.name)
assertDuplicatesInArray(predicates)

// replicate assertAttributePredicateNamesDoNotMatch()
assertDuplicatesInArray(attributes.concat(predicates))
```

The function could also take a `proofRequest` and get the arrays from there, I have no preference really.

Reply:
I like it! Will implement it. 
One question:  `if(arrayLength === uniquearrayLength) return;` on this line you say that if the arrays are evenly long, it is fine, but there could still be a duplicate right?

Reply:
When you create a Set in Javascript from an array it will remove all the duplicates instances of items. like so

```ts
const arr = [1,2,3,4,4,4,4];
const setOfArray = new Set(arr);

console.log(setOfArray);
// output: Set(1,2,3,4);
```

So if the array has a length of lets say 5 and the Set aswell it is safe to assume that there are no duplicates removed when creating the set.